lambda_backend/route_microservice/src/getRoutesByUser.ts

```diff
@@ -58,18 +62,37 @@ const getRoutesByUser: Handler = async (event: GetRoutesByUserEvent) => {
       const gymItems = response.Items as GymItem[];
       gymItems.forEach((gym) => (gymLocationToNameMapping[gym.gymLocation] = gym.gymName));
     } catch (error) {
-      logger.error('getRoutesByUser error', {
-        data: { username, countryCode, error: error.stack },
+      logger.error('getRoutesByUser query gyms error', {
+        data: { routeOwnerUsername, countryCode, error: error.stack },
       });
       throw createError(500, 'Error querying table', error);
     }
   }
 
+  let username = '';
+  if (Authorization) {
+    // Only utilizes token for username, no enforcing authorization as endpoint is GET & public
+    ({ username } = (await jwt_decode(Authorization.split(' ')[1])) as JwtPayload);
+    logger.info('getRoutesByUser user identifier included', { data: { username } });
+  }
   const Items = userRouteIndexItems.map((item) => {
-    return { ...item, gymName: gymLocationToNameMapping[item.gymLocation] };
+    let hasVoted = false;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { upvotes, ...returnItem } = item;
+    upvotes.forEach((name) => {
```

Comment:
Would it be better to just use `Array.find()`?

Reply:
`code quality: A`